aten/src/ATen/core/type.cpp

```diff
@@ -585,6 +585,22 @@ bool is_null_or_equal(c10::optional<T> a, c10::IntArrayRef b) {
   return !a.has_value() || a.value() == b;
 }
 
+
+
+ bool matchStrides(TensorType* ttp, const at::Tensor& t) {
+   TORCH_INTERNAL_ASSERT(t.is_contiguous());
+   auto ssize = *ttp->stride_properties().size();
+   const auto& t_strides = t.strides();
+   for (size_t i = 0; i < ssize; i++) {
+     const auto& stride = *ttp->stride_properties()[i];
+     if (t_strides[*stride.stride_index_] != *stride.stride_ || *stride.contiguous_ != true) {
```

Comment:
Does this fail on any `stride` that isn't `contiguous`?  Is that correct?

Reply:
Not every `ttp->stride_properties()[i];` will have a value either.

Reply:
Yes. We could probably make `matchStrides` support checks on non-contiguous tensors as well and get rid of computeStrideProps completely until we have dynamic shapes. I thought handling contiguous tensors would be a good start on which we can expand later